app/src/androidTest/java/net/skyscanner/backpack/calendar/BpkCalendarTest.kt

```diff
@@ -0,0 +1,180 @@
+package net.skyscanner.backpack.calendar
+
+import android.view.View
+import android.widget.FrameLayout
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.res.ResourcesCompat
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.action.ViewActions
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.ActivityTestRule
+import net.skyscanner.backpack.BpkSnapshotTest
+import net.skyscanner.backpack.R
+import net.skyscanner.backpack.calendar.model.CalendarRange
+import net.skyscanner.backpack.calendar.presenter.BpkCalendarController
+import net.skyscanner.backpack.demo.MainActivity
+import org.hamcrest.CoreMatchers
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.*
+
+private class BpkCalendarControllerImpl(
+  override val isRtl: Boolean,
+  override val locale: Locale,
+  private val initialStartDate: Calendar? = null,
+  private val initialEndDate: Calendar? = null
+): BpkCalendarController() {
+
+  override val startDate: Calendar
+    get() = initialStartDate ?: super.startDate
+
+  override val endDate: Calendar
+    get() = initialEndDate ?: super.endDate
+
+  override fun onRangeSelected(range: CalendarRange) {}
+
+  override fun isToday(year: Int, month: Int, day: Int): Boolean {
+    return day == 2 && month == 0 && year == 2019
```

Comment:
What is month `0`?

Reply:
January, date apis generally start from 0 to 11 🤷‍♂️

Reply:
Not in the new java.time.* library; there it has a constant value of 1. But you are right, what we are using here is java.util.Calendar which still has the old 0-based starting index. 

https://developer.android.com/reference/java/util/Calendar#JANUARY (available since API 26; added in JDK 1.8)
https://developer.android.com/reference/java/time/Month#JANUARY (available since API 1)

🤷‍♂️ 😅

Reply:
So all in all, you were right. My bad.